### PR TITLE
New ErrorView

### PR DESCRIPTION
**Pull request by DmitrijOkeanij**
- Added ErrorView  
- Many ErrorView constructors and factory methods for different error cases
- Refactor all errors on server and client on new ErrorView
- Make central place on Client to show error messages `src/boot/errorNotify.js`
- Make russian localization on all errors `src/i18n/errors.js`
- Added global ErrorHandler on server

### CODE REVIEW

File: SunEngine/Startup.cs [hunk 1]
```diff
@@ -82,7 +87,7 @@ public void ConfigureServices(IServiceCollection services)
                 .AddJsonFormatters(options =>
                 {
                     options.DateTimeZoneHandling = DateTimeZoneHandling.Utc;
-                   // options.NullValueHandling = NullValueHandling.Ignore;
+                    // options.NullValueHandling = NullValueHandling.Ignore;
```
Comment (PutienkoEN) [inline SunEngine/Startup.cs:90]: The commented code should be removed.

File: SunEngine.Commons/Services/ServiceResult.cs [hunk 2]
```diff
@@ -1,4 +1,5 @@
 using SunEngine.Commons.Controllers;
+using SunEngine.Commons.Misc;
```
Comment (PutienkoEN) [inline SunEngine.Commons/Services/ServiceResult.cs:2]: Is it required? Same goes for other changes in imports where code haven't changed.

File: SunEngine.Commons/Misc/ErrorView.cs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    {
+        public string Code { get; }
+
+        public string Description { get; }
+
+        public string Message { get; set; }
+
+        public string StackTrace { get; set; }
+
+        public bool IsSoft { get; }
```
Comment (PutienkoEN) [inline SunEngine.Commons/Misc/ErrorView.cs:118]: I thought we discussed that flag in ErrorObject it's not required.
Stack trace must not be returned to the client, you sure it should be here?
What the difference between message and description?

File: SunEngine.Commons/Managers/AuthManager.cs [hunk 4]
```diff
@@ -121,17 +104,10 @@ public virtual async Task<ServiceResult> RegisterAsync(NewUserArgs model)
                             $"Please confirm your account by clicking this <a href=\"{emailConfirmUrl}\">link</a>."
                         );
                     }
-                    catch (Exception e)
+                    catch (Exception exception)
```
Comment (PutienkoEN) [inline SunEngine.Commons/Managers/AuthManager.cs:107]: Can we use specific exceptions? Using generic exceptions considered the bad practice.

File: SunEngine.Commons/Filters/IpSpamProtectionFilter.cs [hunk 5]
```diff
@@ -3,58 +3,56 @@
...
     public class IpSpamProtectionFilter : ActionFilterAttribute
     {
         private const string CacheKeyStart = "RFIP";
-        
+
         public int TimeoutSeconds
         {
             set => timeout = TimeSpan.FromSeconds(value);
-            get => (int)timeout.TotalSeconds;
+            get => (int) timeout.TotalSeconds;
```
Comment (PutienkoEN) [inline SunEngine.Commons/Filters/IpSpamProtectionFilter.cs:19]: Is it safe to cast it?

File: SunEngine.Commons/Controllers/PulseController.cs [hunk 6]
```diff
@@ -37,5 +37,13 @@ public virtual async Task<IActionResult> PulseDb()
             bool anyUserGroup = await  db.Roles.AnyAsync();
             return Ok(new {db_UserGroups_AnyAsync = anyUserGroup});
         }
+        
+        [HttpGet]
+        [HttpPost]
+        [AllowAnonymous]
+        public virtual IActionResult TestException()
+        {
+            throw new Exception("TestException");
```
Comment (PutienkoEN) [inline SunEngine.Commons/Controllers/PulseController.cs:46]: Could you explain what it's supposed to do?

File: SunEngine.Commons/Cache/Services/CacheKeyGenerator.cs [hunk 7]
```diff
@@ -1,7 +1,7 @@
 using System.Collections.Generic;
 using System.Linq;
 
-namespace SunEngine.Commons.Cache
+namespace SunEngine.Commons.Cache.Services
```
Comment (PutienkoEN) [inline SunEngine.Commons/Cache/Services/CacheKeyGenerator.cs:4]: Why we have changes to cache here? It's not relevant to this pr.

File: LICENSE.md [hunk 8]
```diff
@@ -6,15 +6,15 @@
...
 Распространять код можно только целиком и без модификаций или обрезаний, и только для некоммерческого использования.
 
-#### Коммерческое использование
-
-Регулируются личным договором между пользователем услуг и Дмитрием Поляниным.
-
 #### Права
 
-Права на весь код проекта принадлежат Дмитрию Полянину.  
+Права на весь код проекта принадлежат Дмитрию Полянину (далее основателю проекта).
```
Comment (PutienkoEN) [inline LICENSE.md:11]: This is not related to current PR.

**Review by PutienkoEN — CHANGES_REQUESTED** [review]

File: SunEngine.Commons/Misc/ErrorView.cs (hunk 3 above)
Comment (DmitrijOkeanij) [reply SunEngine.Commons/Misc/ErrorView.cs:118]: Message is message to client user, or Exception.Message

File: SunEngine.Commons/Filters/IpSpamProtectionFilter.cs (hunk 5 above)
Comment (DmitrijOkeanij) [reply SunEngine.Commons/Filters/IpSpamProtectionFilter.cs:19]: We can cast double to int, its normal

File: SunEngine.Commons/Controllers/PulseController.cs (hunk 6 above)
Comment (DmitrijOkeanij) [reply SunEngine.Commons/Controllers/PulseController.cs:46]: Just for testing time for testing new Exception system. After merge and tests it can be deleted.

File: SunEngine.Commons/Cache/Services/CacheKeyGenerator.cs (hunk 7 above)
Comment (DmitrijOkeanij) [reply SunEngine.Commons/Cache/Services/CacheKeyGenerator.cs:4]: + I need to think about how work with PR. Not clear for me now
